tests/test_mariadb.py

```diff
@@ -276,3 +280,76 @@ def test_mariadb_healthcheck_galera_cluster_disabled(auto_container_per_test):
     _wait_for_server(conn)
 
     conn.run_expect([1], "healthcheck.sh --su-mysql --galera_online")
+
+
+_DB_ENV = {
+    "MARIADB_USER": _OTHER_DB_USER,
+    "MARIADB_PASSWORD": _OTHER_DB_PW,
+    "MARIADB_DATABASE": _TEST_DB,
+    "MARIADB_ROOT_PASSWORD": MARIADB_ROOT_PASSWORD,
+    "MARIADB_AUTO_UPGRADE": "1",
+}
+
+
+@pytest.mark.parametrize("ctr_image", MARIADB_CONTAINERS)
+@pytest.mark.skipif(
+    OS_VERSION in ("15.5",), reason="MariaDB upgrade scenario not supported"
+)
+def test_mariadb_upgrade(
+    container_runtime: OciRuntimeBase,
+    pytestconfig: pytest.Config,
+    ctr_image: DerivedContainer,
+    tmp_path: Path,
+) -> None:
+    mounts = [BindMount(host_path=tmp_path, container_path="/var/lib/mysql")]
+    mariadb_old = DerivedContainer(
+        base="registry.suse.com/suse/mariadb:10.6",
```

Comment:
There's no guarantee that you will be able to migrate from 10.6 to newer MariaDB containers.

I believe the test should include a mapping from which version to which newer version, otherwise this test could fail on a new container image, not because it is broken, but because MariaDB does not support migrating 10.6 to version X.Y.

So instead of `MARIADB_CONTAINERS` it should be passed something like `MARIADB_CONTAINERS_UPGRADE_VERSIONS`, which would be a tuple, or another map style type.

Reply:
@alexandrevicenzi as per the docs here: https://mariadb.com/kb/en/upgrading-between-major-mariadb-versions/ we should be able to upgrade between any 2 major versions of mariadb.

Reply:
Right now, the test case written only to verify the upgrade scenario for SP6, as the URL for old mariadb container is hard coded in the test case we are yet to customise the test case to choose the url for old container based on OS_VERSION.
Will update the test case.

Reply:
right now the test case will run for only 15.6.